src/components/OrganizationContainer/OrganizationContainer.tsx

```diff
@@ -0,0 +1,258 @@
+import { gql, useQuery } from '@apollo/client'
+import { Row, Alert } from 'react-bootstrap'
+import { useQueryState } from 'next-usequerystate'
+
+import Error from "../Error/Error"
+import Pager from "../Pager/Pager"
+import { Organization, OrganizationRecord } from '../Organization/Organization';
+import { OrganizationMetadataRecord } from '../OrganizationMetadata/OrganizationMetadata';
+import { DoiType } from '../DoiContainer/DoiContainer'
+import DoiMetadata from '../DoiMetadata/DoiMetadata'
+
+type Props = {
+    rorId: string;
+};
+
+interface OrganizationResult {
+    id: string;
+    name: string;
+    alternateName: string[];
+    url: string;
+    address: {
+        country: string
+    }
+    identifiers: [{
+        identifier: string,
+        identifierType: string
+    }];
+    works: Works;
+}
+
+interface Works {
+    totalCount: number
+    resourceTypes: ContentFacet[]
+    pageInfo: PageInfo
+    published: ContentFacet[]
+    nodes: DoiType[]
+}
+
+interface ContentFacet {
+    id: string
+    title: string
+    count: number
+}
+
+interface PageInfo {
+    endCursor: string
+    hasNextPage: boolean
+}
+
+interface OrganizationQueryData {
+    organization: OrganizationResult
+}
+
+interface OrganizationQueryVar {
+    id: string;
+    cursor: string;
+}
+
+export const ORGANIZATION_GQL = gql`
+query getOrganizationQuery($id: ID!, $cursor: String) {
+    organization(id: $id) {
+        id,
+        name,
+        alternateName,
+        url,
+        address {
+        country
+        },
+        identifiers {
+            identifier,
+            identifierType,
+        },
+        works(first: 25, after: $cursor) {
+            totalCount
+            pageInfo {
+              endCursor
+              hasNextPage
+            }
+            resourceTypes {
+              title
+              count
+            }
+            published {
+              title
+              count
+            }
+            nodes {
+              doi
+              id
+              titles{
+                title
+              }
+              types{
+                resourceTypeGeneral
+                resourceType
+              }
+              creators {
+                id
+                name
+                givenName
+                familyName
+              }
+              version
+              publicationYear
+              publisher
+              descriptions {
+                description
+              }
+              rights {
+                rights
+                rightsUri
+                rightsIdentifier
+              }
+              fieldsOfScience {
+                id
+                name
+              }
+              language {
+                id
+                name
+              }
+              registrationAgency {
+                id
+                name
+              }
+              registered
+              rights {
+                rights
+              }
+              citationCount
```

Comment:
I think there are a few fields here that you don't need for the fields item. What do you think? Maybe you will make your qquery faster if you don't request them

Reply:
Yes you're right, I think this needs a further iteration to work out exactly what we need on the organization page